src/status_im2/common/nickname_drawer/view.cljs

```diff
@@ -0,0 +1,84 @@
+(ns status-im2.common.nickname-drawer.view
+  (:require [clojure.string :as string]
+            [quo.design-system.colors :as colors]
+            [quo2.components.icon :as icons]
+            [quo2.core :as quo]
+            [react-native.core :as rn]
+            [reagent.core :as reagent]
+            [status-im2.common.nickname-drawer.style :as style]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn bad-nickname?
+  [nickname]
+  (string/blank? @nickname))
+
+(defn add-nickname-toast
+  [primary-name entered-nickname public-key]
+  (fn []
+    (rf/dispatch [:hide-bottom-sheet])
+    (rf/dispatch [:toasts/upsert
+                  {:id         :add-nickname
+                   :icon       :correct
+                   :icon-color (:positive-01 @colors/theme)
+                   :text       (i18n/label
+                                :t/set-nickname-toast
+                                {:primary-name primary-name
+                                 :nickname     (string/trim @entered-nickname)})}])
+    (rf/dispatch [:contacts/update-nickname public-key (string/trim @entered-nickname)])))
+
+(defn nickname-drawer
+  [{:keys [title description contact accessibility-label
+           close-button-text]}]
+  (let [{:keys [primary-name nickname public-key]} contact
+        entered-nickname                           (reagent/atom nickname)
+        photo-path                                 (when-not (empty? (:images contact))
+                                                     (rf/sub [:chats/photo-path public-key]))]
+    (fn []
+      [rn/view
+       {:style               style/nickname-container
+        :accessibility-label accessibility-label}
+       [quo/text
+        {:weight :semi-bold
+         :size   :heading-1} title]
+       [rn/view {:style (style/context-container)}
+        [quo/user-avatar
+         {:full-name        primary-name
+          :profile-picture  photo-path
+          :size             :xxs
+          :status-indicator false}]
+        [quo/text
+         {:weight :medium
+          :size   :paragraph-2
+          :style  {:margin-left 4}} primary-name]]
+       [quo/input
+        {:type              :text
+         :blur?             true
+         :placeholder       (i18n/label :t/type-nickname)
+         :auto-focus        true
+         :max-length        32
+         :on-change-text    (fn [nickname]
+                              (reset! entered-nickname nickname))
+         :on-submit-editing (add-nickname-toast primary-name entered-nickname public-key)}]
+       [rn/view
+        {:style style/nickname-description-container}
+        [icons/icon :i/info
+         {:size 16}]
+        [quo/text
+         {:weight :regular
+          :size   :paragraph-2
+          :style  (style/nickname-description)}
+         description]]
+       [rn/view {:style style/buttons-container}
+        [quo/button
+         {:type     :grey
+          :style    {:flex 0.48}
```

Comment:
can you elaborate on why use `:flex 0.48`?

Reply:
It's a very specific value to make the confirmation drawer at the bottom have the same spacing between buttons as design.
Like here status-im2.common.confirmation-drawer.view

Reply:
hmm, you should put this in a named variable.. we are using this in other parts of the application too? Then it should be re-used across all uses. Or consider adding a comment about it's use.

Reply:
also I think the designs are quite similar to this buttons used in request to join community drawer:
https://github.com/status-im/status-mobile/blob/develop/src/status_im2/contexts/communities/menus/request_to_join/view.cljs#L60-L79

can you take a look and see if you can use a similar approach as I did not need to use magic values there.

Reply:
The magic value is also used in `status-im2.common.confirmation-drawer.view`, I think it'd be better to solve both together in a followup

Reply:
sure, whatever you think is best. I pushed a commit with these changes if you want to include it here:
d164393140418a62de73fad5296d5b6103b79d55

or else it's 

style.cljs
```
(def buttons-container
  {:margin-top      20
   :flex-direction  :row
   :align-items     :center
   :justify-content :space-evenly})

(def cancel-button
  {:flex         1
   :margin-right 12})

(def action-button
  {:flex 1})
  ```
  
  view.cljs
  ```